utils/manifest-utils/src/keystore.rs

```diff
@@ -142,6 +143,14 @@ impl Clone for Box<dyn Keystore> {
     }
 }
 
+pub trait SignatureVerifier {
+    /// Signature digest algorithm
+    fn with_alg(&mut self, alg: &str) -> Box<&mut dyn SignatureVerifier>;
+    /// Verifies `signature` of given `data`.
+    /// Returns Ids of certificates issuing signing certificate starting from root-most certificate.
+    fn verify(&self, data: &str, signature: &str) -> anyhow::Result<Vec<String>>;
```

Comment:
Wanted to make it one-shot method taking `self`, but got defeated by ownership model.

Reply:
It does not matter now. If Audited Payload for Golem cert will contain only one property (with merged cert, signature and algorithm) this Trait will probably be removed.